sources/api/netdog/src/net_config/test_macros/bonding.rs

```diff
@@ -0,0 +1,81 @@
+macro_rules! bonding_tests {
+    ($version:expr) => {
+        mod bonding {
+            use $crate::net_config::deserialize_config;
+            use $crate::net_config::test_macros::gen_boilerplate;
+
+            gen_boilerplate!($version, "bonding");
+
+            #[test]
+            fn ok_config() {
```

Comment:
Generating test code like this confounds IDE tooling. Just something to keep in mind in the future. One cannot step through code or click on the green "run this test" button with generated code.

That being said, I love the fact that so much testing is taking place and the test configuration files serve as excellent documentation.